src/Middleware/DispatchMiddleware.php

```diff
@@ -40,6 +78,15 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
 
         $middleware = $routeResult->getMatchedMiddleware();
 
+        if (! $middleware instanceof ServerMiddlewareInterface) {
+            $middleware = $this->prepareMiddleware(
+                $middleware,
+                $this->router,
+                $this->responsePrototype,
+                $this->container
+            );
```

Comment:
No. This call only gets executed if we do not have an http-interop/http-middleware instance. If `route()` was called (either directly, or indirectly via `__call()`), then this logic will be skipped.

Essentially, this is a fallback for if non-interop middleware was injected in a `Route` instance, which will only happen if you're creating those instances manually.